libarchive/archive_read_support_format_rar.c

```diff
@@ -3097,17 +3102,17 @@ copy_from_lzss_window_to_unp(struct archive_read *a, const void **buffer,
   }
 
   windowoffs = lzss_offset_for_position(&rar->lzss, startpos);
-  if(windowoffs + length <= lzss_size(&rar->lzss)) {
+  if(windowoffs + length <= (size_t)lzss_size(&rar->lzss)) {
```

Comment:
These casts to `size_t` are safe and they only widen the value of `lzss->mask`. `mask` is always `rar->dictionary_size - 1`, and `parse_codes` uses `int` throughout to store them. `dictionary_size` is used as an argument to `realloc`, and so must be positive.